apps/users-service/src/routes/authRoutes.ts

```diff
@@ -0,0 +1,107 @@
+import { Request, Response } from 'express';
+const express = require('express');
+const bcrypt = require('bcryptjs');
+const { query } = require('../database/connection');
+
+const router = express.Router();
+
+// POST /api/auth/signup
+router.post('/signup', async (req: Request, res: Response) => {
+  try {
+    const { email, password, firstName, lastName } = req.body;
+
+    // Basic validation
+    if (!email || !password) {
+      return res.status(400).json({ error: 'Email and password are required' });
+    }
+
+    // Simple email format check
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      return res.status(400).json({ error: 'Invalid email format' });
+    }
+
+    // Password length check
+    if (password.length < 6) {
+      return res
+        .status(400)
+        .json({ error: 'Password must be at least 6 characters long' });
+    }
+
+    // Hash password
+    const saltRounds = 12;
+    const passwordHash = await bcrypt.hash(password, saltRounds);
```

Comment:
Avoid using "magic numbers", declare this in a const at the top level of the file:
```suggestion
    const SALT_ROUNDS = 12; // declare at top level
    const passwordHash = await bcrypt.hash(password, SALT_ROUNDS);
```

Reply:
✅ Added SALT_ROUNDS and DUPLICATE_KEY_ERROR_CODE constants